CorsixTH/Src/th_movie.cpp

```diff
@@ -132,20 +132,20 @@ void movie_picture_buffer::allocate(SDL_Renderer* pRenderer, int iWidth,
   // read_index is only used in this thread.
   read_index = write_index;
 
-  std::lock_guard<std::mutex> lock(mutex);
+  std::scoped_lock lock(mutex);
   picture_count = 0;
   allocated = true;
   cond.notify_one();
 }
 
 void movie_picture_buffer::deallocate() {
   {
-    std::lock_guard<std::mutex> lock(mutex);
+    std::scoped_lock lock(mutex);
     allocated = false;
   }
 
   for (movie_picture& picture : picture_queue) {
-    std::lock_guard<std::mutex> pictureLock(picture.mutex);
+    std::scoped_lock lock(mutex);
     picture.deallocate();
   }
```

Comment:
I am guessing this is intended?

It uses a different mutex, but as it seems to deallocate the object, I can see that can be tricky.

Reply:
Line 180 still uses the picture lock. Should not be the same lock?